src/conv.jl

```diff
@@ -45,6 +57,13 @@ function conv!(y::AbstractArray{T,3}, x::AbstractArray{T,3}, w::AbstractArray{T,
     return y
 end
 
+function crossconv!(y::AbstractArray{T,3}, x::AbstractArray{T,3}, w::AbstractArray{T,3};
+               pad = 0, stride = 1, dilation = 1) where T
+    args = map(x -> reshape(x, size(x,1),1,size(x,2),size(x,3)), (y, x, w))
+    crossconv!(args..., pad = (pad...,0), stride = (stride...,1), dilation = (dilation...,1))
```

Comment:
That's `crosscor`. It should also be a thin wrapper that doesn't need as much duplication as you've added here; if you just call `conv!` here with the right mode, for example, you won't need specific 2D and 3D wrappers.

Reply:
But `conv!` doesn't take the `mode` parameter in its arguments at the moment, (and you also advised me not to expose mode in `conv!`).

Reply:
I think it would be ok to have it in `conv!` (perhaps as `flipkernel`) but just not document it. That makes it a bit easier to support `crosscor!` without a lot of duplication. You can also then avoid the gradient wrappers, since we can just support flipkernel in Flux.